setup_yarn.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -e
+
+require_var() {
+  if [ -z "$1" ]; then
+    >&2 echo "$2"
+    return 1
+  fi
+  return 0
+}
+
+require_env() {
+  require_var "$2" "Missing env var '$1'"
+}
+
+setup_yarn() {
+  if [ "$ARTIFACTORY_YARN_SETUP" == "false" ]; then
+    echo "Skipping yarn setup because ARTIFACTORY_YARN_SETUP=$ARTIFACTORY_YARN_SETUP"
+    return 0
+  fi
+
+  require_env "ARTIFACTORY_USERNAME" "$ARTIFACTORY_USERNAME" || return 1
+  require_env "ARTIFACTORY_TOKEN" "$ARTIFACTORY_TOKEN" || return 1
+
+  local yarnrc
+  yarnrc="$HOME/.yarnrc.yml"
```

Comment:
Does `yarnrc` file support scopes like NPM does?

Reply:
it does, but as discussed under readme.md, i don't think its beneficial for our use cases